weave-js/src/core/serverApi.ts

```diff
@@ -27,6 +27,15 @@ export interface ServerAPI {
     assetPath: string
   ): Promise<DirMetadata | FileMetadata | null>;
 
+  getArtifactMembershipFileMetadata?(
```

Comment:
Defining this as optional for now so that the fe code can compile without failure (the method is implemented for serverApiProd in a [downstream PR](https://github.com/wandb/core/pull/27819)). 

In a later PR, where this method is called by a weave-op, we make this a required method: https://github.com/wandb/weave/pull/3815/files#diff-76b7ae5a4d57358be592b41ca3e31102ebe8ce589b127f258cb7c8c5b4af45c3